### PR TITLE
Add configure option for gpfdist and make gpfdist through top level makefile

### PR DESCRIPTION
**Pull request by Chibin**
We would like to add gpfdist as part of the default build for GPDB OSS.

We have also added a --disable-gpfdist option for people who does not want to build it.

### CODE REVIEW

**Comment by yydzero** [comment]
Looks like CI is broken?

**Comment by Chibin** [comment]
Yes, it looks like travis CI is using a higher version of gcc (4.6.3), which exposed some unused values. Will be taking a look.

**Comment by hlinnaka** [comment]
Hmm. The reason gpfdist lives in a separate directory and has its own autoconf script is that it has a bunch of extra dependencies that the server doesn't have. At least libapr IIRC. How do you pass extra options to the gpfdist configure?

**Comment by Chibin** [comment]
@hlinnaka That's a good point.  
I was thinking of getting rid of the configure file and let gpfdist just build through gpdb, but the configure file is too good to remove... so I'm planning on passing the "--with-apr-config" property through the top level configure and use it on the gpfdist configure. Otherwise, people can just use --disable-gpfdist.
Would that make sense?
We also want to have "--enable-transformations" to be set by default as well, I'm not sure as to why it would be off.

**Comment by Chibin** [comment]
I wanted to add gpfdist as part of the installcheck-good/installcheck in the top level and check if gpfdist is enabled, but it might be too intrusive...?

File: gpAux/Makefile [hunk 1]
```diff
@@ -357,7 +349,7 @@ define BUILD_STEPS
 	cd $(BUILDDIR) && $(MAKE) $(parallelexec_opts) install
 	#@$(MAKE) greenplum_path INSTLOC=$(INSTLOC)
 	#@$(MAKE) groupSession INSTLOC=$(INSTLOC)
-	@$(MAKE) mkgpfdist INSTCFLAGS=$(INSTCFLAGS)
+	#@$(MAKE) mkgpfdist INSTCFLAGS=$(INSTCFLAGS)
 	@$(MAKE) mkdbgen INSTLOC=$(INSTLOC)
```
Comment (yydzero) [inline gpAux/Makefile]: Is above line still used?  If not, prefer to delete instead of comment out.

File: gpAux/Makefile [hunk 2]
```diff
@@ -1336,7 +1328,7 @@ clean :
 	@if [ -d Release -a -f Release/GNUmakefile ]; then cd Release; $(MAKE) clean; fi
 	@if [ -d $(GPPGDIR) -a -f $(GPPGDIR)/GNUmakefile ]; then cd $(GPPGDIR); $(MAKE) clean; fi
 	@if [ -d $(GPMGMT)/bin ]; then cd $(GPMGMT)/bin; $(MAKE) clean; fi
-	@if [ -d extensions/gpfdist ]; then cd extensions/gpfdist; $(MAKE) clean; fi
+	#@if [ -d extensions/gpfdist ]; then cd extensions/gpfdist; $(MAKE) clean; fi
 	@if [ -d $(GPPERFMON) ]; then cd $(GPPERFMON); $(MAKE) clean; fi
```
Comment (yydzero) [inline gpAux/Makefile]: same, if not used, please remove instead of comment out.

File: gpAux/extensions/Makefile [hunk 3]
```diff
@@ -3,18 +3,53 @@
...
+
+ext_dir=../ext
+
+top_builddir = ../..
+include $(top_builddir)/src/Makefile.global
+
+ENV_JAVA_HOME:=$(shell echo $$JAVA_HOME)
+ifneq (${ENV_JAVA_HOME},)
+	JAVA_HOME=${ENV_JAVA_HOME}
+endif
```
Comment (yydzero) [inline gpAux/extensions/Makefile]: Do we need Java for gpfdist?

**Pull request merged**